fiat-google-groups/src/main/java/com/netflix/spinnaker/fiat/roles/google/GoogleDirectoryUserRolesProvider.java

```diff
@@ -211,12 +262,15 @@ protected Groups getGroupsFromEmail(String email) throws IOException {
     return groups;
   }
 
-  private GoogleCredential getGoogleCredential() {
+  private GoogleCredentials getGoogleCredential() {
     try {
-      if (StringUtils.isNotEmpty(config.getCredentialPath())) {
-        return GoogleCredential.fromStream(new FileInputStream(config.getCredentialPath()));
+      if (StringUtils.isNotEmpty(config.getCredentialPath())
+          && StringUtils.isNotEmpty(config.getAdminUsername())) {
+        return ServiceAccountCredentials.fromStream(new FileInputStream(config.getCredentialPath()))
```

Comment:
FYI:  Watch this as well... GoogleCredential vs. ServiceAccountCredential...

Reply:
That said... given the scoping... this MAY be fine.  This is one I don't know the GCP stuff AS WELL ... so asking :)